### PR TITLE
Editable mde link

### PR DESCRIPTION
**Pull request by sashasashasasha151**
## Related issue(s)

https://issues.serokell.io/issue/EDNA-117

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [ ] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [ ] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [ ] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: backend/src/Edna/Library/Web/API.hs [hunk 1]
```diff
@@ -177,6 +177,15 @@ data CompoundEndpoints route = CompoundEndpoints
       :> ReqBody '[JSON] URI
       :> Put '[JSON] (WithId 'CompoundId CompoundResp)
 
+  , -- | Edit ChemSoft link for the compound with given ID
```
Comment (gromakovsky) [inline backend/src/Edna/Library/Web/API.hs]: s/ChemSoft/MDe

File: backend/src/Edna/Library/Web/Types.hs [hunk 2]
```diff
@@ -136,6 +136,8 @@ data CompoundResp = CompoundResp
   -- ^ Name of the compound, it may be changed to be a number later.
   , crChemSoft :: Maybe URI
   -- ^ Link to ChemSoft.
+  , crMde :: Maybe URI
+  -- ^ Link to ChemSoft.
```
Comment (gromakovsky) [inline backend/src/Edna/Library/Web/Types.hs]: s/ChemSoft/MDe

File: backend/test/Test/DB/Gen.hs [hunk 3]
```diff
@@ -60,6 +60,7 @@ genCompoundRec compoundId = do
   cName <- genName
   cAdditionDate <- genLocalTime
   cChemsoftLink <- show <<$>> Gen.maybe genURI
+  cMdeLink <- show <<$>> Gen.maybe genURI
```
Comment (gromakovsky) [inline backend/test/Test/DB/Gen.hs]: Hm, perhaps it's more correct to use `renderURI` here and one line above.

File: frontend/src/components/ContextActions/ContextItems.tsx [hunk 4]
```diff
@@ -14,10 +14,11 @@ export type ContextItemType = "edit" | "delete" | "metadata" | "rename" | "prima
 
 interface ContextItemProps {
   type: ContextItemType;
+  value?: string;
   onClick: () => void;
 }
 
-export function ContextItem({ type, onClick }: ContextItemProps): React.ReactElement {
+export function ContextItem({ type, value, onClick }: ContextItemProps): React.ReactElement {
```
Comment (gromakovsky) [inline frontend/src/components/ContextActions/ContextItems.tsx:21]: Is it done to make it possible to specify custom text in menu items?

**Review by gromakovsky — APPROVED** [review]
I've tested it, seems to work fine. I have some minor comments, but aside from that LGTM.

File: frontend/src/components/ContextActions/ContextItems.tsx (hunk 4 above)
Comment (sashasashasasha151) [reply frontend/src/components/ContextActions/ContextItems.tsx:21]: Yes

File: frontend/src/components/dialogs/AddLinkDialog.tsx [hunk 5]
```diff
@@ -80,12 +81,21 @@ export function AddLinkDialog({ target }: AddLinkDialogProps): React.ReactElemen
   const refreshCompounds = useCompoundsRefresher();
   const editing =
     (target.kind === "methodology" && target.object.item.confluence) ||
-    (target.kind === "compound" && target.object.item.chemSoft);
+    (target.kind === "compound-chemsoft" && target.object.item.chemSoft) ||
+    (target.kind === "compound-mde" && target.object.item.mde);
   const editingLink =
-    target.kind === "methodology" ? target.object.item.confluence : target.object.item.chemSoft;
```
Comment (pva701) [inline frontend/src/components/dialogs/AddLinkDialog.tsx:85]: I didn't like this code before (when I wrote it), now I don't like it's even more :putya:

**Review by pva701 — APPROVED** [review]

**Pull request merged**